Rocket/Rocket/Extensions/StringExtension.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  StringExtension.swift
+//  Rocket
+//
+//  Created by Adela Mišicáková on 19.07.2022.
+//
+
+import Foundation
+
+extension String {
+    
+    //MARK: - Formate dateString
+    // - Formates dateString from format yyyy-MM-dd to d.M.yyyy
+    // - Returns formatted dateString
+    
+    func formateDateString () -> String {
+        let inputDateFormatter = DateFormatter()
+        
+        inputDateFormatter.dateFormat = "yyyy-MM-dd"
+        let date = inputDateFormatter.date(from: self)!
+        
+        let outputDateFormatter = DateFormatter()
+        outputDateFormatter.dateFormat = "d.M.yyyy"
```

Comment:
Ak ti tie Stringy chodia z API tak by si sa nikdy nemala spoliehať na to, že ti prídu v správnom formáte. Ak náhodou z API príde niečo divné tak ti crashne appka kvôli force unwrapingu 🙂 Takže toto sa treba určite zbaviť a druhá vec je, že vytváranie DateFormatterov je časovo náročné a nie je dobré ich vytvárať dva krát pri každom volaní tejto funkcie. Dala by som ich ako lazy properties do tejto extension a vo funkcii ich iba použila.

Reply:
https://sarunw.com/posts/how-expensive-is-dateformatter/